File: source/clibCore/Interfaces/cpApi/CPContextClass.vb

Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
Imports Contensive.BaseClasses
Imports System.Runtime.InteropServices

Namespace Contensive.Core
    '
    ' comVisible to be activeScript compatible
    '
    <ComVisible(True)>
    <ComClass(CPContextClass.ClassId, CPContextClass.InterfaceId, CPContextClass.EventsId)>
    Public Class CPContextClass
        Implements IDisposable
        '
#Region "COM GUIDs"
        Public Const ClassId As String = "9FC3B58E-F6A4-4DEA-BE39-B40B09FBE0B7"
        Public Const InterfaceId As String = "4CD84EB3-175C-4004-8811-8257849F549A"
        Public Const EventsId As String = "8C6AC359-68B4-49A3-A3BC-7A53CA16EA45"
#End Region
        '
        Protected disposed As Boolean = False
        Private cp As CPClass
        Private localallowProfileLog As Boolean = False
        '
        '
        ' Constructor - Initialize the Main and Csv objects
        '
        Public Sub New(ByRef cpParent As CPClass)
            MyBase.New()
            cp = cpParent
        End Sub
        '
        ' dispose
        '
        Protected Overridable Overloads Sub Dispose(ByVal disposing As Boolean)
            If Not Me.disposed Then
                Call appendDebugLog(".dispose, dereference main, csv")
                If disposing Then
                    '
                    ' call .dispose for managed objects
                    '
                End If
                '
                ' Add code here to release the unmanaged resource.
                '
                Me.disposed = True
            End If
        End Sub
        '
        '
        Public Property allowDebugLog() As Boolean
            Get
                Return cp.core.debug_allowDebugLog
            End Get
            Set(ByVal value As Boolean)
                cp.core.debug_allowDebugLog = value
            End Set
        End Property
        ''
        '' appName is set as the argument of init( appName ) - not really a context
        ''
        'Public Property appName() As String
        '    Get
        '        Return localappName
        '    End Get
        '    Set(ByVal value As String)
        '        localappName = value
        '    End Set
        'End Property
        '
        '
        '
        Public Property pathPage() As String
            Get
                Return cp.core.webServer.requestPathPage
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestPathPage = value
            End Set
        End Property
        '
        '
        '
        Public Property referrer() As String
            Get
                Return cp.core.webServer.requestReferrer
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestReferrer = value
            End Set
        End Property
        '
        '
        '
        Public Property domain() As String
            Get
                Return cp.core.webServer.requestDomain
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestDomain = value
            End Set
        End Property
        '
        '
        '
        Public Property queryString() As String
            Get
                Return cp.core.webServer.requestQueryString
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestQueryString = value
            End Set
        End Property
        '
        '
        '
        Public Property isSecure() As Boolean
            Get
                Return cp.core.webServer.requestSecure
            End Get
            Set(ByVal value As Boolean)
                cp.core.webServer.requestSecure = value
            End Set
        End Property
        '
        '
        '
        Public Property remoteIp() As String
            Get
                Return cp.core.webServer.requestRemoteIP
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestRemoteIP = value
            End Set
        End Property
        '
        '
        '
        Public Property browserUserAgent() As String
            Get
                Return cp.core.webServer.requestBrowser
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestBrowser = value
            End Set
        End Property
        '
        '
        '
        Public Property acceptLanguage() As String
            Get
                Return cp.core.webServer.requestLanguage
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestLanguage = value
            End Set
        End Property
        '
        '
        '
        Public Property accept() As String
            Get
                Return cp.core.webServer.requestHttpAccept
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestHttpAccept = value
            End Set
        End Property
        '
        '
        '
        Public Property acceptCharSet() As String
            Get
                Return cp.core.webServer.requestHttpAcceptCharset
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestHttpAcceptCharset = value
            End Set
        End Property
        '
        '
        '
        Public Property profileUrl() As String
            Get
                Return cp.core.webServer.requestHttpProfile
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestHttpProfile = value
            End Set
        End Property
        '
        '
        '
        Public Property xWapProfile() As String
            Get
                Return cp.core.webServer.requestxWapProfile
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestxWapProfile = value
            End Set
        End Property
        '
        '
        '
        Public Property isBinaryRequest() As Boolean
            Get
                Return cp.core.webServer.requestFormUseBinaryHeader
            End Get
            Set(ByVal value As Boolean)
                cp.core.webServer.requestFormUseBinaryHeader = value
            End Set
        End Property
        '
        '
        '
        Public Property binaryRequest() As Byte()
            Get
                Return cp.core.webServer.requestFormBinaryHeader
            End Get
            Set(ByVal value As Byte())
                cp.core.webServer.requestFormBinaryHeader = value
            End Set
        End Property
        '
        '====================================================================================================
        ''' <summary>
        ''' get or set request cookies using a name value string similar to a querystring 
        ''' </summary>
        ''' <returns></returns>
        Public Property cookies() As String
            Get
                Dim cookieString As String = ""
                For Each kvp As KeyValuePair(Of String, iisController.cookieClass) In cp.core.webServer.requestCookies
                    cookieString &= "&" & kvp.Key & "=" & kvp.Value.value
                Next
                If cookieString.Length > 0 Then
                    cookieString.Substring(1)
                End If
                Return cookieString
            End Get
            Set(ByVal value As String)
                Dim ampSplit As String()
                Dim ampSplitCount As Integer
                If value <> "" Then
                    ampSplit = Split(value, "&")
                    ampSplitCount = UBound(ampSplit) + 1
                    For ampSplitPointer = 0 To ampSplitCount - 1
                        Dim newCookie As New iisController.cookieClass
                        Dim cookieName As String
                        With newCookie
                            Dim NameValue As String = ampSplit(ampSplitPointer)
                            Dim ValuePair As String()
                            ValuePair = Split(NameValue, "=")
                            cookieName = DecodeResponseVariable(CStr(ValuePair(0)))
                            .name = cookieName
                            If UBound(ValuePair) > 0 Then
                                .value = DecodeResponseVariable(CStr(ValuePair(1)))
                            End If
                        End With
                        If cp.core.webServer.requestCookies.ContainsKey(cookieName) Then
                            cp.core.webServer.requestCookies.Remove(cookieName)
                        End If
                        cp.core.webServer.requestCookies.Add(cookieName, newCookie)
                    Next
                End If
            End Set
        End Property
        '
        '
        '
        Public Property form() As String
            Get
                Return genericController.convertNameValueDictToREquestString(cp.core.webServer.requestFormDict)
            End Get
            Set(ByVal value As String)
                cp.core.webServer.requestFormDict.Clear()
                If (Not String.IsNullOrEmpty(value)) Then
                    Dim keyValuePairs As String() = value.Split(CChar("&"))
                    For Each keyValuePair As String In keyValuePairs
                        If (Not String.IsNullOrEmpty(keyValuePair)) Then
                            Dim keyValue As String() = keyValuePair.Split(CChar("="))
                            If keyValue.Length > 1 Then
                                cp.core.webServer.requestFormDict.Add(keyValue(0), keyValue(1))
                            Else
                                cp.core.webServer.requestFormDict.Add(keyValue(0), "")
                            End If
                        End If
                    Next
                End If
            End Set
        End Property
        '==========================================================================================
        ''' <summary>
        ''' A URL encoded name=value string that contains the context for uploaded files. Each file contains five nameValues. The names are prefixed with a counter. The format is as follows: 0formname=formname&0filename=filename&0type=fileType&0tmpFile=tempfilename&0error=errors&0size=fileSize
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <Obsolete("Deprecated", True)>
        Public Property formFiles() As String
            Get
                Return "" 'cp.core.webServer.requesFilesString
            End Get
            Set(ByVal value As String)
                'cp.core.webServer.requesFilesString = value
            End Set
        End Property
        '
        '
        '
        Public Property requestNameSpaceAsUnderscore() As Boolean
            Get
                Return cp.core.webServer.requestSpaceAsUnderscore
            End Get
            Set(ByVal value As Boolean)
                cp.core.webServer.requestSpaceAsUnderscore = value
            End Set
        End Property
        '
        '
        '
        Public Property requestNameDotAsUnderscore() As Boolean
            Get
                Return cp.core.webServer.requestDotAsUnderscore
            End Get
            Set(ByVal value As Boolean)
                cp.core.webServer.requestDotAsUnderscore = value
            End Set
        End Property

        '
        '
        '
        Public ReadOnly Property responseRedirect() As String
            Get
                Return cp.core.webServer.bufferRedirect
            End Get
        End Property
        '
        '
        '
        Public ReadOnly Property responseBuffer() As String
            Get
                Return cp.core.doc.docBuffer
            End Get
        End Property
        '
        '
        '
        Public ReadOnly Property responseContentType() As String
            Get
                Return cp.core.webServer.bufferContentType
            End Get
        End Property
        '
        '
        '
        Public ReadOnly Property responseCookies() As String
            Get
                Return cp.core.webServer.bufferCookies
            End Get
        End Property
        '
        '
        '
        Public ReadOnly Property responseHeaders() As String
            Get
                Return cp.core.webServer.bufferResponseHeader
            End Get
        End Property
        '
        '
        '
        Public ReadOnly Property responseStatus() As String
            Get
                Return cp.core.webServer.bufferResponseStatus
            End Get
        End Property
        '
        '
        ' append to logfile
        '
        Private Sub appendDebugLog(ByVal copy As String)
            'My.Computer.FileSystem.WriteAllText("c:\clibCpDebug.log", Now & " - cp.content, " & copy & vbCrLf, True)
            ' 'My.Computer.FileSystem.WriteAllText(System.AppDocmc.main_CurrentDocmc.main_BaseDirectory() & "cpLog.txt", Now & " - " & copy & vbCrLf, True)
        End Sub
        '
        ' testpoint
        '
        Private Sub tp(ByVal msg As String)
            'Call appendDebugLog(msg)
        End Sub
#Region " IDisposable Support "
        ' Do not change or add Overridable to these methods.
        ' Put cleanup code in Dispose(ByVal disposing As Boolean).
        Public Overloads Sub Dispose() Implements IDisposable.Dispose
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
        Protected Overrides Sub Finalize()
            Dispose(False)
            MyBase.Finalize()
        End Sub
#End Region
    End Class
End Namespace

File: source/core/Views/CPContextClass.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Processor;
using Contensive.Processor.Models.Db;
using Contensive.Processor.Controllers;
using static Contensive.Processor.Controllers.genericController;
using static Contensive.Processor.constants;
//
namespace Contensive.Processor {
    public class CPContextClass : IDisposable {
        //
        #region COM GUIDs
        public const string ClassId = "9FC3B58E-F6A4-4DEA-BE39-B40B09FBE0B7";
        public const string InterfaceId = "4CD84EB3-175C-4004-8811-8257849F549A";
        public const string EventsId = "8C6AC359-68B4-49A3-A3BC-7A53CA16EA45";
        #endregion
        //
        protected bool disposed { get; set; } = false;
        private CPClass cp { get; set; }
        private bool localallowProfileLog { get; set; } = false;
        //
        //==========================================================================================
        //
        public CPContextClass( CPClass cpParent) : base() {
            cp = cpParent;
        }
        //
        //==========================================================================================
        //
        protected virtual void Dispose(bool disposing) {
            if (!this.disposed) {
                if (disposing) {
                    //
                    // call .dispose for managed objects
                    //
                }
                //
                // Add code here to release the unmanaged resource.
                //
                this.disposed = true;
            }
        }
        //
        //==========================================================================================
        //
        public bool allowDebugLog {
            get {
                return cp.core.doc.allowDebugLog;
            }
            set {
                cp.core.doc.allowDebugLog = value;
            }
        }
        //
        //==========================================================================================
        //
        public string pathPage {
            get {
                return cp.core.webServer.requestPathPage;
            }
            set {
                cp.core.webServer.requestPathPage = value;
            }
        }
        //
        //==========================================================================================
        //
        public string referrer {
            get {
                return cp.core.webServer.requestReferrer;
            }
            set {
                cp.core.webServer.requestReferrer = value;
            }
        }
        //
        //==========================================================================================
        //
        public string domain {
            get {
                return cp.core.webServer.requestDomain;
            }
            set {
                cp.core.webServer.requestDomain = value;
            }
        }
        //
        //==========================================================================================
        //
        public string queryString {
            get {
                return cp.core.webServer.requestQueryString;
            }
            set {
                cp.core.webServer.requestQueryString = value;
            }
        }
        //
        //==========================================================================================
        //
        public bool isSecure {
            get {
                return cp.core.webServer.requestSecure;
            }
            set {
                cp.core.webServer.requestSecure = value;
            }
        }
        //
        //==========================================================================================
        //
        public string remoteIp {
            get {
                return cp.core.webServer.requestRemoteIP;
            }
            set {
                cp.core.webServer.requestRemoteIP = value;
            }
        }
        //
        //==========================================================================================
        //
        public string browserUserAgent {
            get {
                return cp.core.webServer.requestBrowser;
            }
            set {
                cp.core.webServer.requestBrowser = value;
            }
        }
        //
        //==========================================================================================
        //
        public string acceptLanguage {
            get {
                return cp.core.webServer.requestLanguage;
            }
            set {
                cp.core.webServer.requestLanguage = value;
            }
        }
        //
        //==========================================================================================
        //
        public string accept {
            get {
                return cp.core.webServer.requestHttpAccept;
            }
            set {
                cp.core.webServer.requestHttpAccept = value;
            }
        }
        //
        //==========================================================================================
        //
        public string acceptCharSet {
            get {
                return cp.core.webServer.requestHttpAcceptCharset;
            }
            set {
                cp.core.webServer.requestHttpAcceptCharset = value;
            }
        }
        //
        //==========================================================================================
        //
        public string profileUrl {
            get {
                return cp.core.webServer.requestHttpProfile;
            }
            set {
                cp.core.webServer.requestHttpProfile = value;
            }
        }
        //
        //==========================================================================================
        //
        public string xWapProfile {
            get {
                return cp.core.webServer.requestxWapProfile;
            }
            set {
                cp.core.webServer.requestxWapProfile = value;
            }
        }
        //
        //==========================================================================================
        //
        public bool isBinaryRequest {
            get {
                return cp.core.webServer.requestFormUseBinaryHeader;
            }
            set {
                cp.core.webServer.requestFormUseBinaryHeader = value;
            }
        }
        //
        //==========================================================================================
        //
        public byte[] binaryRequest {
            get {
                return cp.core.webServer.requestFormBinaryHeader;
            }
            set {
                cp.core.webServer.requestFormBinaryHeader = value;
            }
        }
        //
        //==========================================================================================
        /// <summary>
        /// get or set request cookies using a name value string similar to a querystring 
        /// </summary>
        /// <returns></returns>
        public string cookies {
            get {
                string cookieString = "";
                foreach (KeyValuePair<string, WebServerController.CookieClass> kvp in cp.core.webServer.requestCookies) {
                    cookieString += "&" + kvp.Key + "=" + kvp.Value.value;
                }
                if (cookieString.Length > 0) {
                    cookieString.Substring(1);
                }
                return cookieString;
            }
            set {
                string[] ampSplit = null;
                int ampSplitCount = 0;
                if (!string.IsNullOrEmpty(value)) {
                    ampSplit = value.Split('&');
                    ampSplitCount = ampSplit.GetUpperBound(0) + 1;
                    for (var ampSplitPointer = 0; ampSplitPointer < ampSplitCount; ampSplitPointer++) {
                        WebServerController.CookieClass newCookie = new WebServerController.CookieClass();
                        string cookieName = null;
                        string NameValue = ampSplit[ampSplitPointer];
                        string[] ValuePair = NameValue.Split('=');
                        cookieName = decodeResponseVariable(encodeText(ValuePair[0]));
                        newCookie.name = cookieName;
                        if (ValuePair.GetUpperBound(0) > 0) {
                            newCookie.value = decodeResponseVariable(encodeText(ValuePair[1]));
                        }
                        if (cp.core.webServer.requestCookies.ContainsKey(cookieName)) {
                            cp.core.webServer.requestCookies.Remove(cookieName);
                        }
                        cp.core.webServer.requestCookies.Add(cookieName, newCookie);
                    }
                }
            }
        }
        //
        //==========================================================================================
        //
        public string form {
            get {
                return genericController.convertNameValueDictToREquestString(cp.core.webServer.requestFormDict);
            }
            set {
                cp.core.webServer.requestFormDict.Clear();
                if (!string.IsNullOrEmpty(value)) {
                    string[] keyValuePairs = value.Split('&');
                    foreach (string keyValuePair in keyValuePairs) {
                        if (!string.IsNullOrEmpty(keyValuePair)) {
                            string[] keyValue = keyValuePair.Split('=');
                            if (keyValue.Length > 1) {
                                cp.core.webServer.requestFormDict.Add(keyValue[0], keyValue[1]);
                            } else {
                                cp.core.webServer.requestFormDict.Add(keyValue[0], "");
                            }
                        }
                    }
                }
            }
        }
        //==========================================================================================
        /// <summary>
        /// A URL encoded name=value string that contains the context for uploaded files. Each file contains five nameValues. The names are prefixed with a counter. The format is as follows: 0formname=formname&0filename=filename&0type=fileType&0tmpFile=tempfilename&0error=errors&0size=fileSize
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        [Obsolete("Deprecated", true)]
        public string formFiles {
            get {
                return "";
            }
            set {
                //
            }
        }
        //
        //==========================================================================================
        //
        public bool requestNameSpaceAsUnderscore {
            get {
                return cp.core.webServer.requestSpaceAsUnderscore;
            }
            set {
                cp.core.webServer.requestSpaceAsUnderscore = value;
            }
        }
        //
        //==========================================================================================
        //
        public bool requestNameDotAsUnderscore {
            get {
                return cp.core.webServer.requestDotAsUnderscore;
            }
            set {
                cp.core.webServer.requestDotAsUnderscore = value;
            }
        }
        //
        //==========================================================================================
        //
        public string responseRedirect {
            get {
                return cp.core.webServer.bufferRedirect;
            }
        }
        //
        //==========================================================================================
        //
        [Obsolete("responseBuffer is deprecated. Each method returns its content.")]
        public string responseBuffer {
            get {
                return string.Empty;
            }
        }
        //
        //==========================================================================================
        //
        public string responseContentType {
            get {
                return cp.core.webServer.bufferContentType;
            }
        }
        //
        //==========================================================================================
        //
        public string responseCookies {
            get {
                return cp.core.webServer.bufferCookies;
            }
        }
        //
        //==========================================================================================
        //
        public string responseHeaders {
            get {
                return cp.core.webServer.bufferResponseHeader;
            }
        }
        //
        //==========================================================================================
        //
        public string responseStatus {
            get {
                return cp.core.webServer.bufferResponseStatus;
            }
        }
        //
        //==========================================================================================
        //
        #region  IDisposable Support 
        // Do not change or add Overridable to these methods.
        // Put cleanup code in Dispose(ByVal disposing As Boolean).
        public void Dispose() {
            Dispose(true);
            GC.SuppressFinalize(this);
        }
        ~CPContextClass() {
            Dispose(false);
            //todo  NOTE: The base class Finalize method is automatically called from the destructor:
            //base.Finalize();
        }
        #endregion
    }
}